### PR TITLE
Push Images to Dockerhub & Github Packages

### PR DESCRIPTION
**Pull request by EngHabu**
- [X] Push Docker Images from Travis to DockerHub
- [X] Create two Github Workflows:
   - Pull Requests
      - [X] Push Docker Images (tagged with git sha) to Github Packages
   - Master
      - [X] Bump SemVer Patch version on Master Merge
      - [X] Push Docker Images (tagged with git tag) to Github Packages

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,9 +13,9 @@ jobs:
       script: make docker_build
     - if: fork = false
```
Comment (wild-endeavor) [inline .travis.yml]:
```suggestion
    - if: fork = false AND branch = master
```

Comment (wild-endeavor) [reply .travis.yml]: can also add `tag IS present`

**Comment by wild-endeavor** [comment]
I know you wanted to move everything over to GitHub at one time @EngHabu but I don't mind doing it piecemeal if you're okay with it.  Can we just use the github workflow to push to both the github docker register and dockerhub?

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Push Docker Image to Github Registry
+        uses: whoan/docker-build-with-cache-action@v5
+        with:
+          username: "${{ github.actor }}"
+          password: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (kumare3) [inline .github/workflows/pull_request.yml:15]: the name of the variable is something differnt. This seems wrong?

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        uses: anothrNick/github-tag-action@1.17.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          WITH_V: true
+          DEFAULT_BUMP: patch
+      - name: Push Docker Image to Github Registry
+        uses: whoan/docker-build-with-cache-action@v5
+        with:
+          username: "${{ github.actor }}"
+          password: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (kumare3) [inline .github/workflows/master.yml:26]: same here

**Comment by kumare3** [comment]
One comment, thank you for doing this, we can then move it to boilerplate and replicate

File: .github/workflows/pull_request.yml (hunk 2 above)
Comment (EngHabu) [reply .github/workflows/pull_request.yml:15]: Not sure what do you mean by that :) ? what do you refer to by "variable" and different compared to what?

**Pull request merged**